cfme/tests/services/test_provision_stack.py

```diff
@@ -121,129 +72,148 @@ def stack_data(appliance, provider, provisioning):
                 'SSHLocation': provisioning['ssh_location']
             }
         else:
-            _stack_data = {
+            stack_data = {
                 'stack_name': stackname,
                 'stack_timeout': stack_timeout,
                 'param_virtualMachineName': vm_name,
                 'param_KeyName': stack_prov['key_name']
             }
-    return _stack_data
+    return stack_data
 
 
-def test_provision_stack(appliance, setup_provider, provider, provisioning, catalog, catalog_item,
-                         request, stack_data):
-    """Tests stack provisioning
+@pytest.fixture
+def dialog_name():
+    return 'dialog_{}'.format(fauxfactory.gen_alphanumeric())
 
-    Metadata:
-        test_flag: provision
-    """
-    catalog_item, template = catalog_item
 
-    @request.addfinalizer
-    def _cleanup_vms():
-        clean_up(stack_data, provider)
+@pytest.yield_fixture
+def template(provider, provisioning, dialog_name, stack):
+    template_type = provisioning['stack_provisioning']['template_type']
+    template_name = fauxfactory.gen_alphanumeric()
+    template = OrchestrationTemplate(template_type=template_type,
+                                     template_name=template_name)
+    file = provisioning['stack_provisioning']['data_file']
+    data_file = load_data_file(str(orchestration_path.join(file)))
+    template.create(data_file.read().replace('CFMETemplateName', template_name))
+    template.create_service_dialog_from_template(dialog_name, template.template_name)
+    yield template
+    if stack.exists:
+        stack.retire_stack()
+    if template.exists:
+        template.delete()
+
+
+@pytest.yield_fixture
+def catalog():
```

Comment:
Fixture catalog can be used here 
https://github.com/ManageIQ/integration_tests/blob/master/cfme/fixtures/service_fixtures.py#L21